tests/SpeechRecognizerTests.ts

```diff
@@ -1756,6 +1756,37 @@ describe.each([true])("Service based tests", (forceNodeWebSocket: boolean) => {
                     }
                 });
         }, 100000);
+
+        test("Endpoint URL With Auth Token Bearer added", (done: jest.DoneCallback) => {
+            // tslint:disable-next-line:no-console
+            console.info("Name: Endpoint URL With Auth Token Bearer added");
+
+            const fakeToken: string = createNoDashGuid();
+            const s: sdk.SpeechConfig = sdk.SpeechConfig.fromAuthorizationToken(fakeToken, "westus");
+            objsToClose.push(s);
+
+            const r: sdk.SpeechRecognizer = BuildRecognizerFromWaveFile(s);
+            objsToClose.push(r);
+
+            r.recognizeOnceAsync(
```

Comment:
>r.recognizeOnceAsync [](http://example.com/codeflow?start=12&length=20)

from experience with what broke in previous SDKs...

I would add another test, that calls recognize twice, in a way that the token should expire, and verify that the header is updated with a second and different guid.

becasue that has caused us issues in the past, where speech works untill the first token expires, and then it didn;t refresh, or there was bad latency in getting the new token.

Reply:
This test actually exists in [LongRunning/SpeechRecoAuthTokenRefreshTests.ts](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/master/tests/LongRunning/SpeechRecoAuthTokenRefreshTests.ts). It takes 20 minutes, though, so it is by default disabled in ADO builds. Locally, it passed with this change.